src/main/java/land/oras/OCILayout.java

```diff
@@ -337,6 +335,71 @@ public Referrers getReferrers(LayoutRef ref, @Nullable ArtifactType artifactType
     private void setPath(Path path) {
         this.path = path;
     }
+    /**
+     * Clean up unreferenced blobs in the OCI layout.
+     *
+     * @return the number of blobs deleted
+     * @throws IOException if an I/O error occurs
+     */
+    public int cleanupUnreferencedBlobs() throws IOException {
+        LOG.info("Starting cleanup of unreferenced blobs in OCI layout: {}", path);
+
+        // Collect all referenced digests
+        Set<String> referencedDigests = new HashSet<>();
+        Path indexPath = path.resolve("index.json");
+        if (!Files.exists(indexPath)) {
+            LOG.warn("No index.json found in layout: {}", path);
+            return 0;
+        }
+
+        String indexContent = Files.readString(indexPath);
+        Map<String, Object> index = JsonUtils.fromJson(indexContent, Map.class);
```

Comment:
Why not using concrete object with Index.fromPath(getIndexPath()) instead of map?